frontends/p4/strengthReduction.cpp

```diff
@@ -107,6 +109,10 @@ const IR::Node* DoStrengthReduction::postorder(IR::BOr* expr) {
     auto r = expr->right->to<IR::Cmpl>();
     if (l && r)
         return new IR::Cmpl(new IR::BAnd(expr->srcInfo, l->expr, r->expr));
+    if (hasSideEffects(expr))
+        return expr;
+    if (expr->left->equiv(*expr->right))
+        return expr->left;
```

Comment:
You check for side effects before folding here, but not in the BAnd case above?  They should be consistent.

Reply:
There is a similar check in BAnd as well.